piet-coregraphics/src/ct_helpers.rs

```diff
@@ -131,38 +109,25 @@ impl<'a> Line<'a> {
     }
 
     pub(crate) fn get_string_range(&self) -> CFRange {
-        unsafe { CTLineGetStringRange(self.0.as_concrete_TypeRef()) }
+        self.0.get_string_range()
     }
 
     pub(crate) fn get_typographic_bounds(&self) -> TypographicBounds {
-        let mut out = TypographicBounds::default();
-        let width = unsafe {
-            CTLineGetTypographicBounds(
-                self.0.as_concrete_TypeRef(),
-                &mut out.ascent,
-                &mut out.descent,
-                &mut out.leading,
-            )
-        };
-        out.width = width;
-        out
+        self.0.get_typographic_bounds()
     }
 
     pub(crate) fn get_string_index_for_position(&self, position: CGPoint) -> CFIndex {
-        unsafe { CTLineGetStringIndexForPosition(self.0.as_concrete_TypeRef(), position) }
+        self.0.get_string_index_for_position(position)
     }
 
-    /// return the 'primary' and 'secondary' offsets on the given line that the boundary of the
+    /// Return the 'primary' and offsets on the given line that the boundary of the
     /// character at the provided index.
     ///
+    /// There is a 'secondary' offset that is not returned by the core-text crate.
```

Comment:
Again, the secondary offset is for BiDi. I suggest language along the lines of "the underlying Core Text function also returns a secondary offset, which is useful for BiDi, but since we are not implementing that yet, we ignore it for now."